utils/preferences/src/preferences.rs

```diff
@@ -24,44 +27,56 @@ pub trait PreferenceKey {
 
 #[macro_export]
 macro_rules! preferences {
-    ($name:ident,
-     $resolved_name:ident,
-     {$($key:ident => $pref:ty),*}
+    (
+        $(#[$doc:meta])*
+        $name:ident,
+        $resolved_name:ident,
+        {
+            $(
+                $(#[$key_doc:meta])*
+                $key:ident: $pref:ty
+            ),*
+        }
      ) => (
-        #[derive(Default, Debug)]
+        #[derive(Debug, Clone)]
+        $(#[$doc])*
         #[non_exhaustive]
-        pub struct $name {
-            pub lid: Option<icu_locale_core::LanguageIdentifier>,
+        pub struct $name<'a> {
+            #[doc = concat!("The locale that these `", stringify!($name), "` use.")]
+            pub lid: &'a $crate::preferences::icu_locale_core::LanguageIdentifier,
             $(
+                $(#[$key_doc])*
                 pub $key: Option<$pref>,
             )*
         }
 
+        impl Default for $name<'_> {
+            fn default() -> Self {
+                Self {
+                    lid: {static D: $crate::preferences::icu_locale_core::LanguageIdentifier = $crate::preferences::icu_locale_core::LanguageIdentifier::default(); &D },
+                    $(
+                        $key: None,
+                    )*
+                }
+            }
+        }
+
         #[non_exhaustive]
         #[derive(Debug, Clone)]
+        #[doc = concat!("The resolved version of `", stringify!($name), "`.")]
         pub struct $resolved_name {
-            pub lid: icu_locale_core::LanguageIdentifier,
+            #[doc = concat!("The locale that these `", stringify!($name), "` use.")]
+            pub data_locale: icu_provider::DataLocale,
 
             $(
+                $(#[$key_doc])*
                 pub $key: $pref,
             )*
         }
 
-        impl From<(icu_locale_core::LanguageIdentifier, $resolved_name)> for $resolved_name {
-            fn from(input: (icu_locale_core::LanguageIdentifier, $resolved_name)) -> Self {
-                Self {
-                    lid: input.0,
-                    $(
-                        $key: input.1.$key,
-                    )*
-                }
-            }
-        }
-
-        impl From<icu_locale_core::Locale> for $name {
-            fn from(loc: icu_locale_core::Locale) -> Self {
-
-                let lid = Some(loc.id);
+        impl<'a> From<&'a $crate::preferences::icu_locale_core::Locale> for $name<'a> {
```

Comment:
Discussion: Do we want an impl from both `Locale` and `&Locale`?

Reply:
Please post discussions in the issue